pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -24,7 +24,8 @@ import (
 )
 
 const (
-	jobFinalizer = "job.finalizers.flink.k8s.io"
+	jobFinalizer         = "job.finalizers.flink.k8s.io"
+	shouldUpdateK8Object = true
```

Comment:
This should not be at the global level.

You should be created and returned within handle.

Reply:
This is only meant to be a const whose value is not updated. Instead of having true, false literals spread across handle methods, opted to use a constant that's a bit more explicitly named, but perhaps this is more confusing?

Reply:
Oh I see. Yeah. this is confusing when you do `!`, without `!` this would have been clear.

Also lines https://github.com/lyft/flinkk8soperator/pull/48/files#diff-9d1b35757e5b357ae56fe201c420afb8R138 confused me. Wondering if this variable can be abused as we grow.

Reply:
You can have another variable `applicationUnchanged` or `applicationModified` if you want to use it in Return statements.

Reply:
Thanks for the feedback! Updated constants to `applicationChanged` and `applicationUnchanged` and limited their usage only in return statements.